File: 1. Two Sum/solution_2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> numAndIndex = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; ++i)
        {
            int curr = nums[i];
            int diff = target - curr;

            if (numAndIndex.ContainsKey(diff))
                return new int[] { numAndIndex[diff], i };

            numAndIndex[curr] = i;
        }

        return new int[] { };
    }
}

File: 1. Two Sum/solution_2.js
var twoSum = function(nums, target) 
{
   let numAndIndex = new Map();

    for (let i = 0; i < nums.length; ++i)
    {
        const curr = nums[i];
        const diff = target - curr;

        if (numAndIndex.has(diff))
            return [numAndIndex.get(diff), i];

        numAndIndex.set(curr, i);
    }

   return [];
};